### ISSUE TITLE
simplify billing work flow for prior auths

### ISSUE THREAD

**Issue by stephenwaite** [openemr/openemr#3894]
### Describe the problem

currently have to use the `Miscellaneous Billing Options` form in order for the prior auth number to be included in the claim

### Describe the solution you'd like

rather see a prior auth widget added to the main pt dashboard whereby the prior auth could be linked to an encounter

then the encounter and claims generation could look to see if a prior auth spanned the date of the encounter and should be linked

the prior auth form found in `Forms Administration` as Unregistered could be called from the widget

**Comment by stephenwaite** [comment]
hi @juggernautsei, here's the `issue/enhancement` started a couple of weeks ago

thinking the form could be moved into the codebase proper and  no longer be enabled through forms->administration so have to figure out an upgrade path for the few folks who have used it as encounter based

hoping it will be much like medical problems are in the dashboard that can be linked to encounters (should help upgraders) but doesn't have to be

here's your draft of the table


>     CREATE TABLE `form_prior_auth` (
>       `id` int(11) NOT NULL auto_increment,
>       `pid` int(11) default NULL,
>       `activity` tinyint(4) NOT NULL default '0',
>       `date` datetime NOT NULL default '1900-01-01 12:12:12',
>       `prior_auth_number` varchar(35) default NULL,
>       `comments` varchar(255) default NULL,
>       `expiration` datetime NULL default '1900-01-01 12:12:12',
>       `units` int(5) default NULL,
>       PRIMARY KEY  (`id`)
>     ) ENGINE=InnoDB;
>

**Comment by juggernautsei** [comment]
Here is an idea to make this a module that can be enabled if needed. What do you think about going down that path? Do we need a prior auth id column in the billing table?

**Comment by stephenwaite** [comment]
hi @juggernautsei , not sure about a module since this could be used extensively imho, pretty sure auth # is in `misc_billings_option` table

**Comment by juggernautsei** [comment]
@stephenwaite  The reason that I mentioned it is because it is not used by every type of practice. So, it needs to have the ability to be disabled if not being used.

**Comment by stephenwaite** [comment]
think it's pretty widely used enough to set a global instead

**Comment by juggernautsei** [comment]
Agreed. 
Below is part of the form mark up. The upgrade path in part should be the adding of the two columns to the current table that is in use. 
    

     #IfMissingColumn form_prior_auth expiration
      ALTER TABLE `form_prior_auth` ADD expiration datetime NULL default '1900-01-01 12:12:12'
     #EndIf
     #IfMissingColumn form_prior_auth units
      ALTER TABLE `form_prior_auth` ADD `units` int(5) default NULL
     #EndIf

Then, remove the prior auth from the registration table. That will remove it from the Misc column on the encounter form and remove it from the form manager.   Also, the table.sql and info.txt will have to be moved from the folder to remove it completely from the form manager.  

Now we need to attach it to the demographics stats column.

![image](https://user-images.githubusercontent.com/834958/92908070-e43c9700-f3f3-11ea-9502-2701cd8b40b0.png)

**Comment by juggernautsei** [comment]
Have you started a repo for this project

**Comment by stephenwaite** [comment]
nope, will just be a branch of off master

**Comment by stephenwaite** [comment]
hi @juggernautsei , decided to bring in the prior auth form from contrib and update it a little bit. Let me know what you think.

**Comment by juggernautsei** [comment]
That is a great idea.

Sherwin

**Comment by stephenwaite** [comment]
It's in master now so next step will be to add some logic to encounter to trigger if would like to auto-create for billing, just thinking out loud.

**Comment by juggernautsei** [comment]
auto create which part? Shouldn't the form be updated first? Just asking because I may have missed that part.

**Comment by juggernautsei** [comment]
I just looked, I will have to pull down the master and take a look at the form.

**Comment by stephenwaite** [comment]
will revisit

**Comment by juggernautsei** [comment]
@stephenwaite 
I am back dealing with this form again. I wanted to know is there a reason that the Prior Auth table is not considered in the claim process? Unless the prior auth is put in the misc billing form. It is not used.

**Comment by stephenwaite** [comment]
hi @juggernautsei , it's a work in process that's why mentioned that the next step is to look there

**Comment by juggernautsei** [comment]
Ok @stephenwaite, so this is not stale now.

**Comment by stephenwaite** [comment]
nope but could use a toaster oven

![toaster](https://user-images.githubusercontent.com/5618372/115022022-cb64fa00-9e8a-11eb-9e2c-3227da3ef87d.gif)

**Comment by juggernautsei** [comment]
Let's see if this works here. 

https://images.app.goo.gl/98kVnv6gJw9Ccovy9

**Comment by juggernautsei** [comment]
@stephenwaite for this customization. The client wanted the Prior Auth form located in the dashboard ribbon.
They also added this requirement. I have done this before long ago. @bradymiller I wanted to get your and Stephen's input on this how can this be added to the patient ribbon when enabled in the forms? I plan on releasing this one to the community when it is working. I reworked the form so that it works in both locations. 

![image](https://user-images.githubusercontent.com/834958/115039345-60bdb980-9e9e-11eb-8898-b8f165074dbd.png)

**Comment by stephenwaite** [comment]
hi @juggernautsei , you looking [here](https://github.com/openemr/openemr/blob/master/src/Menu/PatientMenuRole.php#L188)?

**Comment by juggernautsei** [comment]
No, but @stephenwaite, I will put this back on my radar to add more functionality to the feature.

**Comment by stephenwaite** [comment]
ok, me too, thank you @juggernautsei

**Comment by juggernautsei** [comment]
@stephenwaite after reevaluating the direction I was heading. I am going to take a bold step. I am taking an angular class right now. I think this would be a great project to use for skill-building. The form type that is currently being used does not lend itself to all the things that need to be done with the prior auth. Prior auth needs to be a stand-alone feature build in my opinion. It should not be connected or bound to an encounter. It is entered, stored, managed, and retrieved data.  
Also, I will make this a module that can be installed. I have a lot of ambition for this project. I will create a new repo when I have something to show.

**Comment by stephenwaite** [comment]
ok, sounds great @juggernautsei , good luck with the class!
